### PR TITLE
0.2/743 refactor family model to return entitlements and number of entitled kids

### PR DESCRIPTION
**Pull request by charlesstrange2**
for @alexalala 's blade templates.

Added 
- family@eligibleChildrenCount() : returns a count of born,under school age children.
Altered
- family@expecting() : returns a carbon date for the due date.

### CODE REVIEW

File: app/Family.php [hunk 1]
```diff
@@ -167,7 +168,7 @@ public function getNoticeReasons()
     }
 
     /**
-     * Calculates the entitlement
+     * Calculates the entitlement Attrbute
```
Comment (katjam) [inline app/Family.php]: Attribute

File: app/Family.php [hunk 2]
```diff
@@ -176,18 +177,39 @@ public function getEntitlementAttribute()
...
+     * Gets the due date or Null;
+     *
+     * @return mixed
+     */
     public function getExpectingAttribute()
     {
-        // return true if there is a child that has a false 'born' attribute
-
-        $expecting = false;
+        $due = null;
```
Comment (katjam) [inline app/Family.php:187]: So if the baby is born - returns null

File: app/Family.php [hunk 3]
```diff
@@ -176,18 +177,39 @@ public function getEntitlementAttribute()
...
-                $expecting = true;
-                break;
+            if (!$child->born) {
+                $due = $child->dob;
             }
         }
-        return $expecting;
+        return $due;
+
+        /*
```
Comment (katjam) [inline app/Family.php]: delete?

File: app/Family.php [hunk 4]
```diff
@@ -176,18 +177,39 @@ public function getEntitlementAttribute()
...
+                if (!is_null($due)) {
+                    $due = ($due->gt($child->dob)) ? $child->dob : $due;
+                }
+        */
+    }
+
+    /**
+     * Attribute that gets the number of eligible children
+     *
+     * @return mixed
```
Comment (katjam) [inline app/Family.php]: Int?

File: app/Family.php [hunk 5]
```diff
@@ -176,18 +177,39 @@ public function getEntitlementAttribute()
...
+     */
+    public function getEligibleChildrenCountAttribute()
+    {
+        return $this->children->reduce(function ($count, $child) {
```
Comment (katjam) [inline app/Family.php]: Does this cry if there ore 0 children or return 0?

File: database/factories/ModelFactory.php [hunk 6]
```diff
@@ -184,9 +184,9 @@
 });
 
 // Child - unborn
-$factory->state(App\Child::class, 'withUnbornChild', function (Faker\Generator $faker) {
+$factory->defineAs(App\Child::class, 'unbornChild', function (Faker\Generator $faker) {
```
Comment (katjam) [inline database/factories/ModelFactory.php:187]: I was going to suggest that... 👍

File: tests/Unit/Models/FamilyModelTest.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+use App\Family;
+use App\Carer;
+use App\Child;
+use Illuminate\Foundation\Testing\DatabaseMigrations;
+
+class FamilyModelTest extends TestCase
+{
+    use DatabaseMigrations;
+
+    public function setUp()
```
Comment (katjam) [inline tests/Unit/Models/FamilyModelTest.php]: remove unused setup function?

File: tests/Unit/Models/FamilyModelTest.php [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+        // Add some Children
+        $family->children()
+            ->saveMany(
+                collect([
+                    factory(Child::class, 'unbornChild')->make(),
+                    factory(Child::class, 'underOne', 2)->make(),
+                    factory(Child::class, 'underSchoolAge')->make(),
+                    factory(Child::class, 'overSchoolAge')->make(),
+                ])->flatten()
+            );
```
Comment (katjam) [inline tests/Unit/Models/FamilyModelTest.php:32]: neat!

File: tests/Unit/Models/FamilyModelTest.php [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+                collect([
+                    factory(Child::class, 'underOne', 2)->make(),
+                    factory(Child::class, 'underSchoolAge')->make(),
+                    factory(Child::class, 'overSchoolAge')->make(),
+                ])->flatten()
+            );
+
+        // Test we've not got an expecting date
+        $this->assertEquals(null, $family->expecting);
+
```
Comment (katjam) [inline tests/Unit/Models/FamilyModelTest.php]: bit gappy

File: tests/Unit/Models/FamilyModelTest.php [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+        // Add a prenant Family
```
Comment (katjam) [inline tests/Unit/Models/FamilyModelTest.php]: pregnant

File: tests/Unit/Models/FamilyModelTest.php [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+                    factory(Child::class, 'underOne', 2)->make(),
+                    factory(Child::class, 'underSchoolAge')->make(),
+                    factory(Child::class, 'overSchoolAge')->make(),
+                ])->flatten()
+            );
+
+        // Test we've not got an expecting date
+        $this->assertEquals($pregnancy->dob, $pregnant_family->expecting);
+    }
+
```
Comment (katjam) [inline tests/Unit/Models/FamilyModelTest.php]: gappy

**Review by katjam — APPROVED** [review]
's ok I guess. 🐮

File: app/Family.php (hunk 2 above)
Comment (charlesstrange2) [reply app/Family.php:187]: Yup. it's supposed to find and return the due date if it exists.

**Pull request merged**